app/src/main/java/org/thoughtcrime/securesms/database/RecipientDatabase.java

```diff
@@ -350,6 +350,20 @@ public void setBlocked(@NonNull Iterable<Recipient> recipients, boolean blocked)
     notifyRecipientListeners();
   }
 
+  // Delete a recipient with the given address from the database
+  public void deleteRecipient(@NonNull String recipientAddress) {
+    SQLiteDatabase db = getWritableDatabase();
+    db.beginTransaction();
+    try {
+      int rowCount = db.delete(TABLE_NAME, ADDRESS + " = ?", new String[] { recipientAddress });
+      db.setTransactionSuccessful();
+      if (rowCount == 0) { Log.w(TAG, "Could not find to delete recipient with address: " + recipientAddress); }
+    } finally {
+      db.endTransaction();
+    }
```

Comment:
I know this pattern exists in this class elsewhere, but I don't think transaction is needed for a single query. So this could just well be written like this:

```suggestion
    int rowCount = db.delete(TABLE_NAME, ADDRESS + " = ?", new String[] { recipientAddress });      if (rowCount == 0) { Log.w(TAG, "Could not find to delete recipient with address: " + recipientAddress); }
```